subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -1178,7 +1177,8 @@ void setTestExecuter(TestExecuter<JvmTestExecutionSpec> testExecuter) {
     @Nullable
     private String getEffectiveExecutable() {
         if (javaLauncher.isPresent()) {
-            return ((DefaultToolchainJavaLauncher) javaLauncher.get()).getExecutable();
+            // This will fail on Java < 7, however toolchain integration is not planned for Java 6 support
```

Comment:
Out of curiosity, what's the problem with Java < 7 ?

Reply:
`Path` is not available. Was introduced in 7.

Reply:
More important is the fact that this code will only be exercised in the main daemon and not in a worker AFAICT. So I should rephrase this sentence. It is added because the IDE will flag this as an error given the language level on the subproject.